### PR TITLE
[MI-120] Allows the client to use the Groups resource

### PR DESCRIPTION
**Pull request by joseconsador**
Adds the Group resource and unit test

/cc @miogalang @jwswj @mmolina @atroche 
 
### References
 - Jira link: https://zendesk.atlassian.net/browse/MI-120

### Risks
 - None

### CODE REVIEW

File: src/Zendesk/API/HttpClient.php [hunk 1]
```diff
@@ -126,10 +128,10 @@ public function __construct(
         }
 
         $this->subdomain = $subdomain;
-        $this->username  = $username;
-        $this->hostname  = $hostname;
-        $this->scheme    = $scheme;
-        $this->port      = $port;
```
Comment (jwswj) [inline src/Zendesk/API/HttpClient.php]: I feel like this indentation was changed recently? Which one are we using?

**Comment by jwswj** [comment]
:+1:

**Comment by joseconsador** [comment]
:+1:

File: src/Zendesk/API/HttpClient.php (hunk 1 above)
Comment (joseconsador) [inline src/Zendesk/API/HttpClient.php]: We had different spacing schemes set on PHPStorm. I've now matched it to align the equal sign.

**Comment by miogalang** [comment]
:+1:

**Pull request merged**